web/source/osk/visualKeyboard.ts

```diff
@@ -1675,26 +1576,125 @@ namespace com.keyman.osk {
+          return;
+        }
+        this.pendingSubkey = pendingLongpress;
+
+        pendingLongpress.promise.then(function(subkeyPopup) {
+          if(_this.pendingSubkey == pendingLongpress) {
+            _this.pendingSubkey = null;
+          }
+
+          if(subkeyPopup) {
+            // Clear key preview if any
+            _this.showKeyTip(null,false);
+  
+            _this.subkeyGesture = subkeyPopup;
+            subkeyPopup.promise.then(function(keyEvent: text.KeyEvent) {
+              // Allow active cancellation, even if the source should allow passive.
+              // It's an easy and cheap null guard.
+              if(keyEvent) {
+                PreProcessor.raiseKeyEvent(keyEvent);
+              }
+              _this.clearPopup();
+            });
+          }
+        });
+      }
+    }
+
+    /**
+     * Updates all currently-pending and activated gestures.
+     * 
+     * @param currentKey    The key currently underneath the most recent touch coordinate
+     * @param previousKey   The previously-selected key
+     * @param touch         The current touch-coordinate for the gesture
+     * @returns true if should fully capture input, false if input should 'fall through'.
+     */
+    updateGestures(currentKey: KeyElement, previousKey: KeyElement, touch: Touch): boolean {
+      let key0 = previousKey;
+      let key1 = currentKey;
+
+      // Clear previous key highlighting, allow subkey controller to highlight as appropriate.
+      if(this.subkeyGesture) {
+        if(key0) {
+          key0.key.highlight(false);
+        }
+        this.subkeyGesture.updateTouch(touch);
+
+        this.keyPending = null;
+        this.touchPending = null;
+
+        return true;
+      }
+
+      this.currentTarget = null;
+
+      // If popup is visible, need to move over popup, not over main keyboard
+      // Could be turned into a browser-longpress specific implementation within browser.PendingLongpress?
+      // Not completely sure what the correct, generalized abstraction would be for that...
+      // and this PR's already big enough, anyway.
+      if(key1 && key1['subKeys'] != null) {
+        // Show popup keys immediately if touch moved up towards key array (KMEW-100, Build 353)
+        if((this.touchY - touch.pageY > 5) && this.pendingSubkey && this.pendingSubkey instanceof browser.PendingLongpress) {
+          this.pendingSubkey.resolve();
+        }
+      }
+
+      // If there is an active popup menu (which can occur from the previous block),
+      // a subkey popup exists; do not allow base key output.
+      if(this.subkeyGesture) {
+        return true;
+      }
+
+      if(key0 && key1 && (key1 != key0) && (key1.id != '')) {
+        // While there may not be an active subkey menu, we should probably update which base key
+        // is being highlighted by the current touch & start a pending longpress for it.
+        this.clearPopup();
+        this.initGestures(key1, touch);
+      }
```

Comment:
Formerly part of `moveOver`:

https://github.com/keymanapp/keyman/blob/759f3be852be7bc78c35b6a37748691cf3ee5b3f/web/source/osk/visualKeyboard.ts#L790-L792

If the base key is changed AND there is no realized gesture (no active subkey menu), we're allowed to change the base key, and thus the base key used for longpresses.  The easiest way is to simply cancel the old one and start a new one.